distribution/src/agent.js

```diff
@@ -75,75 +101,99 @@ function provideInitialize(data) {
-      if (accruedToDistributedRatio.gt(distributionThresholdPercent)) {
-        findings.push(createDistributionAlert(
-          protocolName,
-          protocolAbbreviation,
-          developerAbbreviation,
-          accruedToDistributedRatio,
-          receiver = transferEvent.args.to,
-          compDistributed = amountCompDistributedBN.toString(),
-          compAccrued = prevBlockCompAccruedBN.toString()
-        ));
+    const transferEvents = txEvent.filterLog(ERC20_TRANSFER_EVENT, compoundTokenAddress).filter(
+      (transferEvent) => transferEvent.args.from.toLowerCase() === comptrollerAddress,
+    );
+    await Promise.all(transferEvents.map(async (transferEvent) => {
+      const amountCompDistributedBN = new BigNumber(transferEvent.args.value.toString()).div(
+        compoundTokenDecimalsMultiplier,
+      );
+
+      // if we don't reach the minimum threshold for distributed COMP
+      //  do not bother calculating the ratio to the previous accrued amount
+      if (amountCompDistributedBN.gt(minimumDistributionAmount)) {
+        // determine Comptroller.compAccrued() in previous block
+        const { blockNumber } = txEvent;
+        const prevBlockCompAccrued = await comptrollerContract.compAccrued(
+          transferEvent.args.to, { blockTag: blockNumber - 1 },
+        );
+        const prevBlockCompAccruedBN = new BigNumber(prevBlockCompAccrued.toString()).div(
+          compoundTokenDecimalsMultiplier,
+        );
+
+        // calculate ratio of accrued to distributed COMP
+        const accruedToDistributedRatio = amountCompDistributedBN.div(
+          prevBlockCompAccruedBN,
+        ).times(100);
+
+        // check against a maximum "sane" distribution amount of COMP
+        //  and alert then even if the previous is zero?
+        if (amountCompDistributedBN.gt(maximumSaneDistributionAmount)) {
+          findings.push(createExceedsSaneDistributionAlert(
+            protocolName,
+            protocolAbbreviation,
+            developerAbbreviation,
+            maximumSaneDistributionAmount,
+            transferEvent.args.to,
+            amountCompDistributedBN.toString(),
+          ));
+        } else if (!prevBlockCompAccruedBN.isZero()
+          && accruedToDistributedRatio.gt(distributionThresholdPercent)) {
+          // if the previous accrual is zero our heuristic of using the ratio will not work
+          //  otherwise if the ratio of between the previous accrued amount to the distribution
+          //  amount exceeds the threshold report a finding
+          findings.push(createExceedsRatioThresholdDistributionAlert(
+            protocolName,
+            protocolAbbreviation,
+            developerAbbreviation,
+            accruedToDistributedRatio,
+            transferEvent.args.to,
+            amountCompDistributedBN.toString(),
+            prevBlockCompAccruedBN.toString(),
+          ));
+        }
```

Comment:
Adding a minimum distributed amount attempts to curtail how often this alert will be fired; however, the primary issue here is that this ratio has no 'ground truth' for what dictates a normal value vs. one that should be alerted on. The ratio doesn't really provide meaningful insight into whether or not COMP is being distributed 'correctly,' so I think we should remove the ratio logic and finding altogether.

Reply:
I created #AE-1325 to fix the heuristic used, I agree that we should remove the ratio. I had thought to do it as part of that ticket but I'm fine with removing it here, I didn't imagine the bot getting re-deployed until the new heuristic is updated though.